com.unity.render-pipelines.high-definition/Editor/AssetProcessors/MaterialPostProcessor.cs

```diff
@@ -16,16 +16,56 @@ class MaterialModificationProcessor : AssetModificationProcessor
     {
         static void OnWillCreateAsset(string asset)
         {
-            if (!asset.ToLowerInvariant().EndsWith(".mat"))
+            if (asset.ToLowerInvariant().EndsWith(".mat"))
+            {
+                MaterialPostprocessor.s_CreatedAssets.Add(asset);
                 return;
+            }
 
-            MaterialPostprocessor.s_CreatedAssets.Add(asset);
+            // For .shadergraph assets, this is tricky since the callback will be for the .meta
+            // file only as we don't create it with AssetDatabase.CreateAsset but later AddObjectToAsset
+            // to the .shadergraph via the importer context.
+            // At the time the meta file is created, the .shadergraph is already present
+            // but Load*AtPAth(), GetMainAssetTypeAtPath() etc. won't find anything.
+            // The GUID is already present though, and we actually use those facts to infer we
+            // have a newly created shadergraph.
+
+            // TODO TOCHECK:
+            // HDMetaData subasset will be included after SG creation anyway so unlike for materials
+            // (cf .mat with AssetVersion in OnPostprocessAllAssets) we dont need to manually add a subasset.
+            // For adding them to MaterialPostprocessor.s_ImportedAssetThatNeedSaving for SaveAssetsToDisk()
+            // to make them editable (flag for checkout), we still detect those here, but not sure this is
+            // helpful as right now, even on re-import, a .shadergraph multijson is not rewritten, so only
+            // /Library side serialized data is actually changed (including the generated .shader that can
+            // also change which is why we run shadergraph reimports), and re-import from the same .shadergraph
+            // should be idempotent.
+            if (asset.ToLowerInvariant().EndsWith($".{ShaderGraphImporter.Extension}.meta"))
+            {
+                var sgPath = System.IO.Path.ChangeExtension(asset, null);
+                var importer = AssetImporter.GetAtPath(sgPath);
+                var guid = AssetDatabase.AssetPathToGUID(sgPath);
+                if (!String.IsNullOrEmpty(guid) && importer == null)
+                {
+                    bool test = true;
+                    MaterialPostprocessor.s_CreatedAssets.Add(sgPath);
+                    return;
+                }
+            }
+
+            // Like stated above, doesnt happen:
+            if (asset.ToLowerInvariant().EndsWith($".{ShaderGraphImporter.Extension}"))
+            {
+                MaterialPostprocessor.s_CreatedAssets.Add(asset);
+                return;
+            }
         }
     }
 
     class MaterialReimporter : Editor
     {
         static bool s_NeedToCheckProjSettingExistence = true;
+        static bool s_AutoReimportProjectShaderGraphsOnVersionUpdate = false;
+        internal static bool s_ReimportShaderGraphDependencyOnMaterialUpdate = true;
```

Comment:
The second setting allows to check reimporting a shadergraph on which depend a material when the later is reimported or migrated. This is useful and independent of the "reimport all" feature (ie the first setting is if the polling function should trigger a ReimportAllHDShaderGraphs).